src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/AzureSqlDatabaseBackupShortTermRetentionPolicyCmdletBase.cs

```diff
@@ -20,22 +27,52 @@ public abstract class AzureSqlDatabaseBackupShortTermRetentionPolicyCmdletBase :
         /// </summary>
         private const string PolicyByInputObjectSet = "PolicyByInputObject";
 
+        /// <summary>
+        /// Parameter set for using a resource Id.
+        /// </summary>
+        private const string PolicyByResourceIdSet = "PolicyByResourceIdSet";
+
         /// <summary>
         /// Gets or sets the Database object to get the policy for.
         /// </summary>
-        [Parameter(ParameterSetName = PolicyByInputObjectSet,
+        [Parameter(
+            ParameterSetName = PolicyByInputObjectSet,
             Mandatory = true,
             ValueFromPipeline = true,
             HelpMessage = "The database object to get the policy for.")]
         [ValidateNotNullOrEmpty]
         public AzureSqlDatabaseModel InputObject { get; set; }
 
+        /// <summary>
+        /// Gets or sets the Database object to get the policy for.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = PolicyByResourceIdSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The short term retention policy resource Id.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = PolicyByResourceServerDatabaseSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
         /// <summary>
         /// Gets or sets the name of the database server to use.
         /// </summary>
         [Parameter(ParameterSetName = PolicyByResourceServerDatabaseSet,
             Mandatory = true,
-            ValueFromPipelineByPropertyName = true,
+            ValueFromPipeline = true,
```

Comment:
Usually the atribute parameter [ValueFromPipeline](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/piping-in-powershell.md#valuefrompipeline) is used to bind by complex types. 
```
public string ServerName { get; set; }
```
For string type you may want to keep using ValueFromPipelineByPropertyName.

Reply:
Agh, I switched this when I was playing around with it and forgot to switch back. Thanks for the catch!